Source/Atomic/Network/Connection.cpp

```diff
@@ -200,6 +205,8 @@ void Connection::SetScene(Scene* newScene)
         scene_->StopAsyncLoading();
         SubscribeToEvent(scene_, E_ASYNCLOADFINISHED, HANDLER(Connection, HandleAsyncLoadFinished));
     }
+
+    SubscribeToEvent(scene_, E_COMPONENTREMOVED, HANDLER(Connection, HandleComponentRemoved));

```

Comment:
This was to solve the issue with Javascript component removal not happening until they were garbage collected. By subscribing to this event we can immediately clear out the replication state for the components and the removal event will then be propagated across the network.